src/coreclr/tests/src/JIT/Directed/IL/PInvokeTail/PInvokeTail.il

```diff
@@ -32,18 +32,20 @@
 {
   .maxstack  1
 
-  IL_0000:  ldstr bytearray(48 65 6C 6C 6F 20 57 6F 72 6C 64 21 00 00 00 00) //ldptr      D_00003000
-  IL_0005:  tail. call       int32 puts(int8*)
-  IL_000a:  ret
+    ldstr bytearray(48 65 6C 6C 6F 20 57 6F 72 6C 64 21 00 00 00 00) //ldptr      D_00003000
```

Comment:
Converting unpinned `System.String` to unmanaged pointer, without pinning, does not sound right. What is this test trying to do?

Reply:
No idea. I believe validate eliding the dispatch to the P/Invoke though a managed function via a tail call. I can update the test.

Reply:
Actually isn't the correct solution here to reference the static array at `D_00003000`? Sigh. I will look into what these tests are actually doing.